src/extras/primitives/getMeshMixin.js

```diff
@@ -11,7 +11,9 @@ Object.keys(shaders.standard.schema).forEach(addMapping);
 
 function addMapping (prop) {
   // To hyphenated.
-  var htmlAttrName = prop.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase();
+  var htmlAttrName;
+  if (components[prop]) { return; }
+  htmlAttrName = prop.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase();
```

Comment:
Seems a bit ad-hoc to apply it only to meshMixin primitives, which are only a handful. We probably want to update the core primitive code to detect this and throw a warning.

The symptom problem is `visible: material.visible` mapping could be renamed to `material-visible: material.visible`.